lambda/shared/src/main/scala/feral/lambda/Context.scala

```diff
@@ -30,8 +30,11 @@ final case class Context(
     logStreamName: String,
     identity: Option[CognitoIdentity],
     clientContext: Option[ClientContext],
-    remainingTime: IO[FiniteDuration]
-)
+    remainingTime: F[FiniteDuration]
+) {
+  def mapK[G[_]](fk: F ~> G): Context[G] =
+    this.copy(remainingTime = fk(this.remainingTime))
```

Comment:
This probably isn't the best approach for bincompat but it's enough for now.

Reply:
The `mapK`? I think that's fine. Anything case classes however is dubious 😆

Reply:
Right, I meant the case class `copy`. We'll probably want to make `Context` itself not a case class, right?

Reply:
Yeah absolutely, `Context` we should make an ordinary class. But the bigger issue for bincompat is our use of case classes for the events/responses for circe derivation ...